w06-ruby-on-rails/d3-dawn-ruby-oop/README.md

```diff
@@ -0,0 +1,196 @@
+#Ruby <3's OOP
+
+##Learning Objectives
+By the end of lesson, students will be able to...
+
+* Distinguish between objects & classes in Ruby
+* Initialize classes
+* Define attributes & instance methods
+* Distinguish between instance and class variables
+
+##Hashes (10m)
+
+Let's create a new Hash
+
+* Hashes are simple key/value stores.
+
+**Challenge:**
+How can I organize my data using key/value pairs in Ruby? Like so:
+
+```ruby
+# hash rocket notation
+{:name=>"Napoleon", :fav_food=>"steak", :skills=>["archery", "combat", "egg farming"]}
+
+# can use anything as a key with hash rocket notation, but probably should not
+x = { ["foo"] => "bar" }
+{ [ "foo" ] => "bar" }
+x[["foo"]]
+#=> "bar"
+x = { {nice: "test"}  => "bar" }
```

Comment:
yeah, this is gonna be confusing
how about:
`{ 'url' => 'http://myblog.wordpress.com', 'author'=> 'Melissa ', 'likes' => 3000 }`